content/en/blog: Add a blog post for GSoC'23: re:Arch Unikraft (#268)
unikraft/docs: content/en/blog/2023-06-22-unikraft-gsoc-plat-rearch.md

+++
title = "re:Arch Unikraft"
date = "2023-06-22T18:00:00+03:00"
author = "Rares Miculescu"

> tags = ["GSoC'23"]

Reviewer: ```suggestion
tags = ["GSoC'23", "plat re-arch"]
```

+++

<img width="100px" src="https://summerofcode.withgoogle.com/assets/media/gsoc-2022-badge.svg" align="right" />

> ## About me

Reviewer: ```suggestion
## About Me
```

> <img width="100px" src="https://summerofcode.withgoogle.com/assets/media/gsoc-2022-badge.svg" align="right" />

Reviewer: ```suggestion
<img width="100px" src="https://summerofcode.withgoogle.com/assets/media/gsoc-2023-badge.svg" align="right" />
```

> I'm Rares Miculescu, a second year bachelor student at Politehnica University of Bucharest.

Reviewer: ```suggestion
I'm [Rareș Miculescu](https://github.com/rares-miculescu), a second year bachelor student at [University POLITEHNICA of Bucharest](https://upb.ro/en/).
```

This is my first open source project that I am part of and I love it.
Outside of Unikraft and university, I am obsessed with cars and Formula 1.

## GSoC'23: re:Arch Unikraft

### Motivation

Unikraft provides specialization and strong modularity by strictly following the "everything is a library" concept.

> Despite this concept, Unikraft should have some architectural changes that can optimize and clearly define drivers and platform support.

Reviewer: ```suggestion
Despite this concept, Unikraft requires architectural changes that can optimize and clearly define drivers and platform support, and, in doing so, reduce maintenance effort.
```

## Summary of Objectives

> The goal of this project is, as said earlier, to optimize and clearly define drivers and platform support.

Reviewer: ```suggestion
The goal of this GSoC'23 project is to optimize and clearly define drivers and platform support.
```

> This can be achieved by solving the tasks from [plat-re-arch](https://github.com/orgs/unikraft/projects/36).

Reviewer: ```suggestion
This can be achieved by solving the tasks from [the `Platform Re-architecting` GitHub project](https://github.com/orgs/unikraft/projects/36).
```

> The main focus is to:

Reviewer: ```suggestion
The main tasks are:
```

> - [Move register definitions into arch](https://github.com/orgs/unikraft/projects/36/views/1?pane=issue&itemId=15412393)

Reviewer: ```suggestion
* [Move register definitions into `arch/`](https://github.com/orgs/unikraft/projects/36/views/1?pane=issue&itemId=15412393)
```

> - [Replace libc types with Unikraft defined](https://github.com/orgs/unikraft/projects/36/views/1?pane=issue&itemId=19037961)

Reviewer: ```suggestion
* [Replace libc data types with Unikraft-specific data types](https://github.com/orgs/unikraft/projects/36/views/1?pane=issue&itemId=19037961)
```

> - [Migrate `plat/drivers` into `drivers/`](https://github.com/orgs/unikraft/projects/36/views/1?pane=issue&itemId=23820110)

Reviewer: ```suggestion
* [Migrate `plat/drivers/` into `drivers/`](https://github.com/orgs/unikraft/projects/36/views/1?pane=issue&itemId=23820110)
```

> - [Introduce `drivers/boot-entry`](https://github.com/orgs/unikraft/projects/36/views/1?pane=issue&itemId=23821458)

Reviewer: ```suggestion
* [Introduce `drivers/boot-entry`](https://github.com/orgs/unikraft/projects/36/views/1?pane=issue&itemId=23821458)
```

> After completing these, we will focus on other tasks.

Reviewer: ```suggestion
After completing these, I will focus on other tasks.
```

## Progress

> So far we managed to move the registers from `plat/x86` into `arch/x86` and we are making the final touches on replacing the libc types.

Reviewer: ```suggestion
So far we managed to move the registers from `plat/x86` into `arch/x86`.
We are making the final touches on replacing the libc types.
```

> So far we managed to move the registers from `plat/x86` into `arch/x86`.

Reviewer: ```suggestion
So far, I managed to move the register definitions from `plat/x86` into `arch/x86`.
```

> ## Next steps

Reviewer: ```suggestion
## Next Steps
```

> We are making the final touches on replacing the libc types.

Reviewer: ```suggestion
We are making the final touches on replacing the libc data types.
```

> The next steps, until the next blog post, will be to finish the PR for the replacement of types and start working on migrating the drivers.

Reviewer: ```suggestion
The next steps will be to finish the PR for the replacement of types and start working on migrating the drivers.
```

Reviewer: Add link to the PR.

> 

Reviewer: Remove extra blank line.

> The next steps will be to finish the [PR for the replacement of types](https://github.com/unikraft/unikraft/pull/954) and start working on migrating the drivers.

Reviewer: ```suggestion
The next steps will be to finish the [PR for the replacement of libc data types](https://github.com/unikraft/unikraft/pull/954) and start working on migrating the drivers.
```
